std/stdio.d

```diff
@@ -3846,14 +3846,84 @@ unittest
     }
 }
 
+// roll our own appender, but with "safe" arrays
+private struct ReadlnAppender
+{
+    import core.stdc.string : memcpy;
+
+    char[] buf;
+    size_t cap;
+    size_t pos;
+    size_t initcap;
+
+    void initialize(char[] b)
+    {
+        buf = b;
+        cap = initcap = buf.capacity;
+        if (cap < buf.length)
+            cap = buf.length;
+        pos = 0;
+    }
+    @property char[] data()
+    {
+        // always shrink/extend the buffer to allow reuse in the next call
+        if (initcap > 0 || cap > buf.length)
+            assumeSafeAppend(buf.ptr[0..pos]);
+        return buf.ptr[0..pos];
```

Comment:
Just a note here, I know that this type is only used here. But having `data` always call `assumeSafeAppend` could be expensive, there is no reason to call it more than once for the same data. But practically speaking, the code only calls `data` right before returning (and destroying the appender struct).